jupyter_sphinx/execute.py

```diff
@@ -257,7 +257,14 @@ def write_notebook_output(notebook, output_dir, notebook_name):
         os.path.join(output_dir, notebook_name + ".ipynb"),
     )
     # Write a script too.
-    ext = notebook.metadata.language_info.file_extension
+    ext = notebook.metadata.get("language_info", {}).get("file_extension", None)
+    if ext is None:
+        ext = ".txt"
+        js.logger.warning(
+            "Notebook code has no file extension metadata, " "defaulting to `.txt`",
+            # TODO correct location
+            # location=document.settings.env.docname,
```

Comment:
@choldgraf if you can check what the correct code is here, then I'm fine with it

Reply:
well the docname is sort-of already available to you, but jupyter-sphinx splits it up into the `output_dir` and the `notebook_name` arguments. `output_dir` will be the path to the output dir + any folders in which the current page is nested. and `notebook_name` is just the `basename` of the current document.

So you could either:

1. Split `output_dir` on the path to the root build folder, and take the folders that come after to pre-pend to `notebook_name` and that would be the correct document
2. Make the `output_dir` argument be a path to the root of the build folder, and the `notebook_name` to be the path relative to the root, and then you've already got the `docname` in the function
3. Explicitly pass `docname` as a new argument to `write_notebook_outputs`

Reply:
done in 41e945a